integrations/va-auth0/src/index.tsx

```diff
@@ -213,78 +252,116 @@ const handleFetchEvent: FetchEventCallback<Auth0RuntimeContext> = async (request
-                        logger.debug(
-                            `Did not receive access token. Error: ${(resp && resp.error) || ''} ${
-                                (resp && resp.error_description) || ''
-                            }`,
-                        );
+                let userInfo;
+                if (includeClaimsInToken) {
+                    // Auth0 returns an opaque access token (i.e., one that doesn't include user or custom claims) when
+                    // exchanging an authorization code for a token—unless an audience (aud parameter) is set to a valid
+                    // Auth0 API application during the authorization request.
+                    // In this case, since we only request a token to verify authentication and generate the VA,
+                    // there is no valid API target to specify as audience.
+                    // As a result, we must call the /userinfo endpoint (an OIDC-compliant endpoint) to retrieve user claims.
+                    //
+                    // An alternative approach would be to request an ID Token using the Implicit Grant flow (with form post).
+                    // However, many authentication providers discourage this method in favor of the Authorization Code flow.
+                    // Additionally, some customers may disable the Implicit Grant flow in their Auth0 application.
+                    // Therefore, retrieving user data via the /userinfo endpoint is a more robust solution.
+                    const auth0TokenData = await auth0TokenResp.json<Auth0TokenResponseData>();
+                    if (!auth0TokenData.access_token) {
                         return new Response('Error: No Access Token found in response from Auth0', {
                             status: 401,
                         });
                     }
-                } else {
-                    return new Response('Error: Either ClientId or Client Secret is missing', {
-                        status: 400,
+
+                    const userInfoURL = `${issuerBaseUrl}/userinfo/`;
+                    const userInfoResp = await fetch(userInfoURL, {
+                        method: 'GET',
+                        headers: { Authorization: `Bearer ${auth0TokenData.access_token}` },
+                    });
+
+                    if (!userInfoResp.ok) {
+                        return new Response('Error: Unable to fetch user info from Auth0', {
+                            status: 401,
+                        });
+                    }
+
+                    userInfo = await userInfoResp.json<Record<string, any>>();
+                }
+
+                try {
+                    const privateKey = context.environment.signingSecrets.siteInstallation;
+                    if (!privateKey) {
+                        return new Response('Error: Missing private key from site installation', {
+                            status: 400,
+                        });
+                    }
```

Comment:
Why is it in the try/catch?

Reply:
You mean just the lines 334-339?  No particular reason. Moving it out.
For the rest (e.g jwt.sign) it can throw and best to catch and return a simpler error.